Trust Model with AS metadata and additional clarification about VCI and AS (#247)
italia/eidas-it-wallet-docs: docs/en/trust.rst

    
    The entries that don't have any reference to a known draft or standard are intended to be defined in this technical reference.

> .. list-table::
>    :widths: 20 20 60
>    :header-rows: 1
> 
>    * - Entity
>      - Metadata type
>      - References
>    * - Trust Anchor
>      - ``federation_entity``
>      - `OIDC-FED`_
>    * - Intermediate
>      - ``federation_entity``
>      - `OIDC-FED`_
>    * - Wallet Provider
>      - ``federation_entity``, ``wallet_provider``
>      - --
>    * - Authorization Server
>      - ``federation_entity``, ``oauth_authorization_server``
>      - `OPENID4VCI`_
>    * - Credential Issuer
>      - ``federation_entity``, ``openid_credential_issuer``, [``oauth_authorization_server``]
>      - `OPENID4VCI`_
>    * - Relying Party
>      - ``federation_entity``, ``wallet_relying_party``
>      - `OIDC-FED`_, `OpenID4VP`_

Author: ```suggestion
.. list-table::
   :widths: 20 20 20 60
   :header-rows: 1

   * - OpenID Entity
     - EUDI Entity
     - Metadata Type
     - References
   * - Trust Anchor
     - Trust Anchor
     - ``federation_entity``
     - `OIDC-FED`_
   * - Intermediate
     - Intermediate
     - ``federation_entity``
     - `OIDC-FED`_
   * - Wallet Provider
     - Wallet Provider
     - ``federation_entity``, ``wallet_provider``
     - --
   * - Authorization Server
     - 
     - ``federation_entity``, ``oauth_authorization_server``
     - `OPENID4VCI`_
   * - Credential Issuer
     - PID Provider, (Q)EAA Provider
     - ``federation_entity``, ``openid_credential_issuer``, [``oauth_authorization_server``]
     - `OPENID4VCI`_
   * - Relying Party
     - Relying Party
     - ``federation_entity``, ``wallet_relying_party``
     - `OIDC-FED`_, `OpenID4VP`_
```

> 

Reviewer: I suggest to clarify here: we should distinguish between Entities, Roles, and Metadata Types. For example, the PID/(Q)EAA Provider is the Entity acting the roles of Authorization Server, Credential Issuer, and, optionally, Relying Party (Wallet or OpenID).

.. note::
    Wallet Provider metadata is defined in the section below.

    `Wallet Solution section <wallet-solution.html>`_. 

.. note::

>     In instances where a Credential Issuer also functions as an Authorization Server, 

Author: ```suggestion
    In instances where a PID/EAA Provider implements both the Credential Issuer and the Authorization Server, 
```

>     it is imperative for the Entity to incorporate both 

Author: ```suggestion
    it MUST incorporate both 
```

    ``oauth_authorization_server`` and ``openid_credential_issuer`` within its metadata types.

>     Other implementations may divide the Credential Issuer from the Authorization Server.

Reviewer: What does it mean practically? IMHO the PID/(Q)EAA Provider MUST have both `oauth_authorization_server` and `openid_credential_issuer`  and MAY have `openid_relying_party` or `wallet_relying_party`.

Author: according to openid4vci, they may have authorization_servers (ARRAY) pointing to the AS issuing access token usable for that VCI instance

Reviewer: We currently don't have the `authorization_server` parameter which is optional. I'm open to consider this option but we should analyze all the implications of this in terms of security. At the current stage I suggest having both `oauth_authorization_server` and `openid_credential_issuer` Metadata in the PID/(Q)EAA Provider Entity Configuration, according to openid4vci spec:

"_If this parameter is omitted, the entity providing the Credential Issuer is also acting as the Authorization Server, i.e., the Credential Issuer's identifier is used to obtain the Authorization Server metadata._ "

Author: I agree, while the authorization_serverS claim should be in the vci metadata anyway

Author: @fmarino-ipzs we cannot force the implementations in deploy multiple endpoints on the same entity. AS and RS are separated entities, therefore deploying them separately in different entity, according to the openid specs, has sense.

>     Other implementations may divide the Credential Issuer from the Authorization Server.

Author: ```suggestion
    Other implementations may divide the Credential Issuer from the Authorization Server, when this happens the Credential Issuer metadata MUST contain the `authorization_servers` parameters, including the Authorization Server unique identifier.
```

     - `OIDC-FED`_
   * - Wallet Provider
     - Wallet Provider
     - ``federation_entity``, ``wallet_provider``
     - --
   * - Authorization Server
     - 
     - ``federation_entity``, ``oauth_authorization_server``
     - `OPENID4VCI`_
   * - Credential Issuer
     - PID Provider, (Q)EAA Provider
     - ``federation_entity``, ``openid_credential_issuer``, [``oauth_authorization_server``]
     - `OPENID4VCI`_
   * - Relying Party
     - Relying Party
     - ``federation_entity``, ``wallet_relying_party``
     - `OIDC-FED`_, `OpenID4VP`_

.. note::
    Wallet Provider metadata is defined in the section below.

    `Wallet Solution section <wallet-solution.html>`_. 

.. note::
    In instances where a PID/EAA Provider implements both the Credential Issuer and the Authorization Server, 
    it MUST incorporate both 
    ``oauth_authorization_server`` and ``openid_credential_issuer`` within its metadata types.
    Other implementations may divide the Credential Issuer from the Authorization Server, when this happens the Credential Issuer metadata MUST contain the `authorization_servers` parameters, including the Authorization Server unique identifier.

>     Furthermore, should there be a necessity for User Authentication by the Credential Issuer,
>     it is mandatory to include the relevant metadata type, either ``openid_relying_party`` 

Author: ```suggestion
    Furthermore, should there be a necessity for User Authentication by the Credential Issuer,
    it could be necessary to include the relevant metadata type, either ``openid_relying_party`` 
```
